pkgs/by-name/ze/zepp-simulator/package.nix

```diff
@@ -0,0 +1,189 @@
+
+  patches = [
+    # Fix for qemu input grab not working with NIXOS_OZONE_WL=1
+    ./0001-force_qemu_x11.patch
+  ];
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+    makeWrapper
+
+    dpkg
+    asar
+
+    # QEMU deps (runtime):
+    glib
+    libgcc
+    libcxx
+    zlib
+    libepoxy
+    libpng
+    libaio
+    xorg.libX11
+    libvterm
+    vte
+    gsasl
+    numactl
+    cyrus_sasl
+    gtk3
+    cairo
+    gdk-pixbuf
+    SDL2
+  ];
+
+  dontBuild = true;
+
+  unpackPhase = ''
+    runHook preUnpack
+
+    dpkg-deb -x $src .
+
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # Create output file strucure
+    mkdir -p $out/{bin,opt,share}
+    cp -r opt/simulator $out/opt
+    cp -r usr/share/icons $out/share
+
+    # Remove unnecessary files
+    rm -rf \
+      $out/usr/share/applications/simulator.desktop \
+      $out/opt/simulator/*.so \
+      $out/opt/simulator/libvulkan.so.1 \
+      $out/opt/simulator/swiftshader \
+      $out/opt/simulator/simulator \
+      $out/opt/simulator/resources/firmware/setup_for_linux.sh
+
+    # De-vendor the qemu binary
+    # XXX: DO NOT DO THIS
+    # As stated by @silver.zepp, it's modified and proprietary
+    # ===
+    # rm -r $out/opt/simulator/resources/firmware/qemu_linux/lib/
+    # ln -sf bin/qemu-system-arm $out/opt/simulator/resources/firmware/qemu_linux/qemu-system-arm
+
+    # Use system electron
+    makeWrapper ${electron_30}/bin/electron $out/bin/simulator \
+      --add-flags "--no-sandbox" \
+      --add-flags $out/opt/simulator/resources/app.asar \
+      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
+
+    # HACK: disable sandbox introduced in Electron 20
+    asar extract $out/opt/simulator/resources/app.asar app_unpacked
+    rm $out/opt/simulator/resources/app.asar
+    sed -i \
+      's|contextIsolation: false,|contextIsolation: false, sandbox: false,|g' \
+      app_unpacked/build/electron/process/side-service.js
+    asar pack app_unpacked $out/opt/simulator/resources/app.asar
+    rm -rf app_unpacked
+
+    runHook postInstall
+  '';
+
+  # ignore missing libsasl2.so.2
+  autoPatchelfIgnoreMissingDeps = true;
+
+  # HACK: Replace libsasl2.so.2 with libsasl2.so.3
+  postFixup = ''
+    patchelf \
+      --replace-needed libsasl2.so.2 libsasl2.so.3 \
+      $out/opt/simulator/resources/firmware/qemu_linux/qemu-system-arm
+    chmod +x $out/opt/simulator/resources/firmware/qemu_linux/qemu-system-arm
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "simulator";
+      genericName = "Simulator";
+      desktopName = "Zepp OS Simulator";
+      comment = "Zepp OS Simulator";
+      noDisplay = false;
+      icon = "simulator";
+      tryExec = "simulator";
+      exec = "simulator %U";
+      startupWMClass = "simulator";
+      categories = [
+        "Application"
+        "Development"
+      ];
```

Comment:
`warning: value "Application;Development" for key "Categories" in group "Desktop Entry" contains a deprecated value "Application"`